mixin/setting-admin@1/mixin.php

```diff
@@ -0,0 +1,211 @@
+ *
+ *   <civix><setting-page-title>My Custom Title</setting-page-title></civix>
+ *
+ * @mixinName setting-admin
+ * @mixinVersion 1.0.0
+ * @since 5.67
+ */
+
+namespace Civi\Mixin\SettingAdminV1;
+
+use Civi;
+
+class About {
+
+  /**
+   * @var \CRM_Extension_MixInfo
+   */
+  private $mixInfo;
+
+  /**
+   * @var \CRM_Extension_Info
+   */
+  private $info;
+
+  /**
+   * @param \CRM_Extension_MixInfo $mixInfo
+   */
+  public static function instance(\CRM_Extension_MixInfo $mixInfo): About {
+    $about = new About();
+    $about->mixInfo = $mixInfo;
+    $about->info = \CRM_Extension_System::singleton()->getMapper()->keyToInfo($mixInfo->longName);
+    return $about;
+  }
+
+  public function getPath(): string {
+    return 'civicrm/admin/setting/' . $this->mixInfo->shortName;
+  }
+
+  public function getPerm(): string {
+    return 'administer ' . $this->mixInfo->shortName;
+  }
+
+  public function getLabel(): string {
+    return $this->info->label ? _ts($this->info->label, ['domain' => $this->info->key]) : $this->info->key;
+  }
+
+  public function getPageTitle(): string {
+    // Changing the title any other way is slightly annoying because you have to override both route+nav.
+    // It might be nice if one (route or menu) reliably inherited its title from the other...
+    if (!empty($this->info->civix['setting-page-title'])) {
+      return $this->info->civix['setting-page-title'];
+      // Could call _ts(..., [domain=>...]), but translation appears to happen at another level,
+      // and double-translation might confuse multilingual.
+    }
+    return ts('%1 Settings', [1 => $this->getLabel()]);
+  }
+
+  public function getRoute(): array {
+    return [
+      'title' => $this->getPageTitle(),
+      'page_callback' => 'CRM_Admin_Form_Generic',
+      'access_arguments' => [['administer CiviCRM', $this->getPerm()], 'or'],
+      'adminGroup' => 'System Settings',
+      'desc' => _ts($this->info->description ?: ''),
+    ];
+  }
+
+  public function getNavigation(): array {
+    return [
+      'label' => $this->getPageTitle(),
+      'name' => sprintf('%s_setting_admin', $this->mixInfo->shortName),
+      'url' => $this->getPath() . '?reset=1',
+      // 'icon' => 'crm-i fa-wrench', // None of the other "System Settings" have icons, so we don't.
+      // 'permission' => ['administer CiviCRM', $this->getPerm()],
+      'permission' => "administer CiviCRM,{$this->getPerm()}",
+      'permission_operator' => 'OR',
+    ];
+  }
+
+}
+
+class Nav {
+
+  /**
+   * Visit all items in the nav-tree.
+   *
+   * @param array $items
+   * @param callable $callback
+   *   function(array &$item): mixed
+   *   To short-circuit execution, the callback should return a non-null value.
+   * @return string|null
+   *   Return NULL by default. If the walk was short-circuited, then return that value.
+   */
+  public static function walk(&$items, callable $callback) {
```

Comment:
is this like run but slower?

Reply:
rofl. Yeah, basically. 

FWIW, for an example of this wording, see https://www.php.net/array_walk or https://www.php.net/array_walk_recursive. It's the same basic idea, except that it's tuned to the properties used by nav-menu (e.g. `$item`, `$item['child']`, `$item['attribute']`).